packages/collector/test/tracing/database/db2/test.js

```diff
@@ -0,0 +1,1117 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const expect = require('chai').expect;
+const { supportedVersion, constants } = require('@instana/core').tracing;
+const testUtils = require('../../../../../core/test/test_util');
+const ProcessControls = require('../../../test_util/ProcessControls');
+const globalAgent = require('../../../globalAgent');
+
+const mochaSuiteFn = supportedVersion(process.versions.node) ? describe : describe.skip;
+const DB_LOCAL_CONN_STR = 'HOSTNAME=localhost;UID=node;PWD=REPLACED;PORT=58885;PROTOCOL=TCPIP';
+const DB_REMOTE_CONN_STR = process.env.CI
+  ? process.env.DB2_CONNECTION_STR.replace(/PWD=.*?(?=;)/, 'PWD=REPLACED')
+  : null;
+
+const CONN_STR = DB_REMOTE_CONN_STR || DB_LOCAL_CONN_STR;
+let DB2_NAME;
+let TABLE_NAME_1;
+let TABLE_NAME_2;
+let TABLE_NAME_3;
+
+// NOTE: DB2 has a limitation of 8 chars
+const getDatabaseName = () => {
+  return 'nodedb';
+};
+
+const generateTableName = () => {
+  const randomStr = Array(8)
+    .fill('abcdefghijklmnopqrstuvwxyz')
+    .map(function (x) {
+      return x[Math.floor(Math.random() * x.length)];
+    })
+    .join('');
+
+  return randomStr;
+};
+
+const verifySpans = (agentControls, controls, options = {}) => {
+  return agentControls.getSpans().then(spans => {
+    if (options.expectSpans === false) {
+      expect(spans).to.be.empty;
+      return;
+    }
+
+    const entrySpan = testUtils.expectAtLeastOneMatching(spans, [
+      span => expect(span.n).to.equal('node.http.server'),
+      span => expect(span.f.e).to.equal(String(controls.getPid())),
+      span => expect(span.f.h).to.equal('agent-stub-uuid')
+    ]);
+
+    if (options.expectNoDb2Span) {
+      expect(spans.length).to.equal(1);
+      return;
+    }
+
+    expect(spans.length).to.equal(options.spanLength || 2);
+
+    if (options.verifyCustom) return options.verifyCustom(entrySpan, spans);
+
+    testUtils.expectAtLeastOneMatching(spans, [
+      span => expect(span.t).to.equal(entrySpan.t),
+      span => expect(span.p).to.equal(entrySpan.s),
+      span => expect(span.n).to.equal('db2'),
+      span => expect(span.k).to.equal(constants.EXIT),
+      span => expect(span.f.e).to.equal(String(controls.getPid())),
+      span => expect(span.f.h).to.equal('agent-stub-uuid'),
+      span => expect(span.data.db2.stmt).to.equal(options.stmt || 'select 1 from sysibm.sysdummy1'),
+      span => expect(span.data.db2.dsn).to.equal(`${CONN_STR};DATABASE=${DB2_NAME}`),
+      span => expect(span.async).to.not.exist,
+      span =>
+        options.error
+          ? expect(span.data.db2.error).to.contain(options.error)
+          : expect(span.data.db2.error).to.not.exist,
+      span => (options.error ? expect(span.ec).to.equal(1) : expect(span.ec).to.equal(0))
+    ]);
+  });
+};
+
+mochaSuiteFn('tracing/db2', function () {
+  const timeout = 30 * 1000;
```

Comment:
_thought:_ We often use timeouts that are different on CI and locally. I feel that with using a cloud based DB2 we might need that here as well. But if 30 seconds is good for both cases, that's fine, too.

Reply:
Yeah db2 is tricky. It takes a longer time to bootstrap.
On CI though we can use a smaller timeout because of we use the cloud, true 👍 
Will change :)